.github/workflows/python_actions.yml

```diff
@@ -0,0 +1,142 @@
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Checkout SupportScripts
+      uses: actions/checkout@v2
+      with:
+        repository: SpiNNakerManchester/SupportScripts
+        path: support
+
+    - name: Checkout Spinnaker Dependencies
+      run: |
+        support/gitclone2.sh https://github.com/SpiNNakerManchester/SpiNNUtils.git
+        support/gitclone2.sh https://github.com/SpiNNakerManchester/SpiNNMachine.git
+        support/gitclone2.sh https://github.com/SpiNNakerManchester/SpiNNMan.git
+        support/gitclone2.sh https://github.com/SpiNNakerManchester/PACMAN.git
+        support/gitclone2.sh https://github.com/SpiNNakerManchester/DataSpecification.git
+        support/gitclone2.sh https://github.com/SpiNNakerManchester/spalloc.git
+        support/gitclone2.sh git://github.com/SpiNNakerManchester/SpiNNFrontEndCommon.git
+        support/gitclone.sh https://github.com/SpiNNakerManchester/sPyNNaker.git
+        support/gitclone.sh https://github.com/SpiNNakerManchester/sPyNNaker8.git
+
+    - name: Install -pip
+      run: |
+        python -m pip install --upgrade pip
+        pip install 'pylint<2.5' flake8
+
+    - name: Install SpiNNUtils
+      run: python setup.py install
+      working-directory: SpiNNUtils
+
+    - name: Install SpiNNMachine
+      run: python setup.py install
+      working-directory: SpiNNMachine
+
+    - name: Install SpiNNMan
+      run: python setup.py install
+      working-directory: SpiNNMan
+
+    - name: Preinstall pacman requirements (2.7 Only)
+      if: matrix.python-version == 2.7
+      run: pip install -r requirements.txt
+      working-directory: PACMAN
+
+    - name: Install PACMAN
+      run: python setup.py install
+      working-directory: PACMAN
+
+    - name: Install DataSpecification
+      run: python setup.py install
+      working-directory: DataSpecification
+
+    - name: Install spalloc
+      run: python setup.py install
+      working-directory: spalloc
+
+    - name: Preinstall SpiNNFrontEndCommon requirements (2.7 Only)
+      if: matrix.python-version == 2.7
+      run: pip install -r requirements.txt
+      working-directory: SpiNNFrontEndCommon
+
+    - name: Install SpiNNFrontEndCommon
+      run: python setup.py install
+      working-directory: SpiNNFrontEndCommon
+
+    - name: Preinstall sPyNNaker requirements (2.7 Only)
+      if: matrix.python-version == 2.7
+      run: pip install -r requirements.txt
+      working-directory: sPyNNaker
+
+    - name: Install sPyNNaker
+      run: python setup.py install
+      working-directory: sPyNNaker
+
+    - name: Preinstall sPyNNaker8 requirements (2.7 Only)
+      if: matrix.python-version == 2.7
+      run: pip install -r requirements.txt
+      working-directory: sPyNNaker8
+
+    - name: Install sPyNNaker8
+      run: python setup.py install
+      working-directory: sPyNNaker8
+
+    - name: Create a spynnaker.cfg
```

Comment:
I don't think this is necessary as there are no tests being run from here?

Reply:
agreed removed